packages/status-screen/src/models/splunk.ts

```diff
@@ -0,0 +1,90 @@
+import '../vendor/splunk.min.js';
+
+export interface SplunkApplicationNamespace {
+    app?: string;
+    owner?: string;
+    sharing?: string;
+}
+
+export interface SplunkPassword {
+    name: string;
+    _properties: {
+        clear_password: string;
+    };
+}
+
+export interface SplunkCollectionItem {
+    key: string;
+    value: string;
+    user: string;
+}
+
+export interface SplunkAppAccessor {
+    reload: () => void;
+}
+
+export interface Stanza {
+    name: string;
+}
+
+export interface SplunkAppsAccessor {
+    fetch: () => SplunkAppsAccessor;
+    item: (applicationName: string) => SplunkAppAccessor;
+}
+
+export interface ConfigurationStanzaAccessor {
+    fetch: () => ConfigurationStanzaAccessor;
+    update: (properties: Record<string, string>) => void;
+    list: () => Array<{ name: string }>;
+    properties: () => Record<string, string>;
+    _properties: Record<string, string>;
+}
+
+export interface ConfigurationFileAccessor {
+    create: (stanzaName: string) => void;
+    fetch: () => ConfigurationFileAccessor;
+    item: (
+        stanzaName: string,
+        properties: SplunkApplicationNamespace
+    ) => ConfigurationStanzaAccessor;
+    list: () => Array<{ name: string }>;
+}
+
+export interface ConfigurationsAccessor {
+    fetch: () => ConfigurationsAccessor;
+    create: (configurationFilename: string) => void;
+    item: (stanzaName: string, properties: SplunkApplicationNamespace) => ConfigurationFileAccessor;
+    list: () => Array<{ name: string }>;
+}
+
+export interface SplunkStoragePasswordAccessors {
+    fetch: () => SplunkStoragePasswordAccessors;
+    item: (applicationName: string) => SplunkAppAccessor;
+    list: () => Array<SplunkPassword>;
+    del: (passwordId: string) => void;
+    create: (params: { name: string; realm: string; password: string }) => void;
+}
+
+export interface SplunkService {
+    configurations: (params: SplunkApplicationNamespace) => ConfigurationsAccessor;
+    apps: () => SplunkAppsAccessor;
+    storagePasswords: () => SplunkStoragePasswordAccessors;
+    get: (splunkUrlPath: string, data: any) => void;
+    post: (
+        splunkUrlPath: string,
+        data: any,
+        callback: (err: { data: string }, response: any) => void
+    ) => void;
+}
+
+export enum PasswordKeys {
+    API_KEY = 'api_key',
+    TENANT_ID = 'tenant_id',
+    INGEST_METADATA_ONLY = 'ingest_metadata_only',
+}
+
+export enum CollectionKeys {
+    START_DATE = 'start_date',
+    LAST_FETCHED = 'timestamp_last_fetch',
+    NEXT_TOKEN = 'next_',
+}
```

Comment:
We have to duplicate all of this from package to package? I wonder if we need to create a common package for these interfaces and other things like utils and whatnot.

Reply:
That's what I want to do after the release, I tried quickly yesterday but didn't find an obvious way to do it